apps/alpha-test-app/src/components/layout-demo/LayoutSettings.vue

```diff
@@ -0,0 +1,165 @@
+<script lang="ts" setup>
+import { computed, ref } from "vue";
+import SingleSettingsGroup from "./SingleSettingsGroup.vue";
+import MultiSettingsGroup from "./MultiSettingsGroup.vue";
+import { watch } from "vue";
+import type { SelectionOption } from "sit-onyx";
+import { OnyxHeadline } from "sit-onyx";
+
+export type Settings = {
+  longPageContent?: boolean;
+  showFlyout?: boolean;
+  forceTooltip?: boolean;
+  showStickyContent?: boolean;
+  showToast?: boolean;
+
+  showSideBar?: boolean;
+  showSideBarCollapse?: boolean;
+  showTempOverlayTransparent?: boolean;
+  showTempOverlay?: boolean;
+
+  detailFooter?: boolean;
+  fullFooter?: boolean;
+
+  showPopover?: boolean;
+  showMobileFlyIn?: boolean;
+  showPageLoader?: boolean;
+  topBarFlyout?: boolean;
+};
+
+export type SettingsSection = "content" | "sideBar" | "footer" | "overlay";
+export type SettingsSections = Record<SettingsSection, Settings>;
+
+const props = withDefaults(
+  defineProps<{
+    modelValue: SettingsSections;
+    horizontal?: boolean;
+    show?: SettingsSection[];
+  }>(),
+  {
+    horizontal: false,
+    show: () => ["content", "sideBar", "footer", "overlay"],
+  },
+);
+
+const emit = defineEmits<{
+  "update:modelValue": [value: SettingsSections];
+}>();
+
+const noneOption = { id: "none", label: "None" };
+
+const contentOptions: SelectionOption<undefined>[] = [
```

Comment:
We could give the generic a default value so we don't need `<undefined>` everywhere
```ts
export type SelectionOption<T = unknown>
```